flink-runtime/src/test/java/org/apache/flink/runtime/webmonitor/threadinfo/JobVertexThreadInfoTrackerTest.java

```diff
@@ -117,71 +126,88 @@ public void testGetThreadInfoStats() throws Exception {
     @Test
     public void testCachedStatsUpdatedAfterRefreshInterval() throws Exception {
-        final Duration threadInfoStatsRefreshInterval2 = Duration.ofMillis(10);
-        final long waitingTime = threadInfoStatsRefreshInterval2.toMillis() + 10;
+        final Duration shortRefreshInterval = Duration.ofMillis(1);
 
-        final int requestId2 = 1;
-        final JobVertexThreadInfoStats threadInfoStats2 =
+        // first entry is in the past, so refresh is triggered immediately upon fetching it
+        final JobVertexThreadInfoStats initialThreadInfoStats =
                 createThreadInfoStats(
-                        requestId2, TIME_GAP, Collections.singletonList(threadInfoSample));
-
+                        Instant.now().minus(10, ChronoUnit.SECONDS),
+                        REQUEST_ID,
+                        Duration.ofMillis(5),
+                        Collections.singletonList(threadInfoSample));
+        final JobVertexThreadInfoStats threadInfoStatsAfterRefresh =
+                createThreadInfoStats(1, TIME_GAP, Collections.singletonList(threadInfoSample));
+
+        // register a CountDownLatch with the cache so we can await refresh of the entry
+        CountDownLatch cacheRefreshed = new CountDownLatch(1);
+        Cache<JobVertexThreadInfoTracker.Key, JobVertexThreadInfoStats> vertexStatsCache =
+                createCache(CLEAN_UP_INTERVAL, new LatchRemovalListener<>(cacheRefreshed));
         final JobVertexThreadInfoTracker<JobVertexThreadInfoStats> tracker =
                 createThreadInfoTracker(
                         CLEAN_UP_INTERVAL,
-                        threadInfoStatsRefreshInterval2,
-                        threadInfoStatsDefaultSample,
-                        threadInfoStats2);
-        doInitialRequestAndVerifyResult(tracker);
+                        shortRefreshInterval,
+                        vertexStatsCache,
+                        initialThreadInfoStats,
+                        threadInfoStatsAfterRefresh);
 
-        // ensure that the previous request "expires"
-        Thread.sleep(waitingTime);
+        // no stats yet, but the request triggers async collection of stats
+        assertFalse(tracker.getVertexStats(JOB_ID, EXECUTION_JOB_VERTEX).isPresent());
+        // block until the async call completes and the first result is available
+        tracker.getResultAvailableFuture().get();
 
-        Optional<JobVertexThreadInfoStats> result =
-                tracker.getVertexStats(JOB_ID, EXECUTION_JOB_VERTEX);
+        // retrieve the entry, triggering the refresh as side effect
+        assertExpectedEqualsReceived(
+                initialThreadInfoStats, tracker.getVertexStats(JOB_ID, EXECUTION_JOB_VERTEX));
 
-        assertExpectedEqualsReceived(threadInfoStats2, result);
+        // wait until the entry is refreshed, with generous buffer
+        assertTrue(cacheRefreshed.await(500, TimeUnit.MILLISECONDS));
```

Comment:
```suggestion
        assertTrue(cacheRefreshed.await());
```
Based on recent discussions we no longer use timeouts in tests.